metrics/server.go

```diff
@@ -0,0 +1,40 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog"
+)
+
+// Server is the http server that will be serving the /metrics request for prometheus.
+type Server struct {
+	server *http.Server
+	log    zerolog.Logger
+}
+
+// NewServer creates a new server that exposes metrics.
+func NewServer(log zerolog.Logger, address string) *Server {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+
+	m := Server{
+		server: &http.Server{
+			Addr:    address,
+			Handler: mux,
+		},
+		log: log,
```

Comment:
might be helpful to differentiate logs
```suggestion
		log: log.With().Str("module", "metrics_server").Logger(),
```